feat(LicenseDataModel): Merge LiceneObligation with Obligation

smrutis1 opened the pull request:
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

 * Which issue is this pull request belonging to and how is it solving it? (*#989*)
 * Did you add or update any new dependencies that are required for your change? -No

Issue: Fixes: #989 

### Suggest Reviewer


### How To Test?

* Need to test the license import and export functionality and adding obligation to licenses

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

JaideepPalit commented on scripts/migrations/030_obligation_field_update.py:
```diff
@@ -0,0 +1,86 @@
+        del entity[''+fieldToBeRemoved+'']
+        if not DRY_RUN:
+            db.save(entity)
+            print 'Removing field name '+fieldToBeRemoved+' done'
```
Maybe more info can be added to this statement like Id.
Current O/P
```
Getting all obligation with field obligationDatabaseIds
found 7 obligation with field obligationDatabaseIds in db!
Removing field name obligationDatabaseIds starts
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
```

smrutis1 replied:
appended the `id`

JaideepPalit reviewed (changes requested):
Kindly look into some minor review comments.
Tested the PR seems to be some issue while adding License obligation from License

![image](https://user-images.githubusercontent.com/56534710/94117509-ea0d9000-fe69-11ea-90e7-fe1ccd679ffc.png)

smrutis1 commented on backend/src/src-licenses/src/test/java/org/eclipse/sw360/licenses/TestLicenseClient.java:
```diff
@@ -12,7 +12,7 @@
 import org.eclipse.sw360.datahandler.thrift.licenses.License;
 import org.eclipse.sw360.datahandler.thrift.licenses.LicenseService;
-import org.eclipse.sw360.datahandler.thrift.licenses.LicenseObligation;
+/*import org.eclipse.sw360.datahandler.thrift.licenses.LicenseObligation;*/
```
done

smrutis1 commented on frontend/sw360-portlet/src/main/resources/content/Language.properties:
```diff
@@ -1138,7 +1138,7 @@ to.be.replaced=To be replaced
 obligation.details=Obligation Details
 obligations=Obligations
-obligations.and.obligations=Todos and Obligations
+#obligations.and.obligations=Todos and Obligations
```
done

smrutis1 commented on libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftUtils.java:
```diff
@@ -55,7 +55,8 @@
             .add(Component.class).add(Release.class) // Component service
-            .add(License.class).add(Obligation.class).add(LicenseObligation.class) // License service
+            .add(License.class).add(Obligation.class)
+//            .add(LicenseObligation.class) // License service
```
done

smrutis1 commented:
Thanks for review. Incorporated the review comments along with that addressed the issue related to blank obligation title after adding obligation to title and added a new column(`Obligation Text`) to make it look more verbose and for easy differentiation.

mcjaeger commented:
(tested on stage)

mcjaeger commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/licenses/LicensesPortlet.java:
```diff
@@ -405,46 +410,55 @@ public void addObligations(ActionRequest request, ActionResponse response) throw
+            } else {
+                List<Obligation> obls = client.getObligationsByIds(Arrays.asList(obligationIds));
+                obls.stream().filter(Objects::nonNull).forEach(obl -> wrapException(() -> {
+                    Obligation oblCopy = obl.deepCopy();
```
that seems to be the only relevant part for the if-else

smrutis1 replied:
refactored the code.

JaideepPalit reviewed (approved):
Code looks good to me
